File: src/main/java/edu/marmara/repository/impl/InstructorRepositoryImpl.java
package edu.marmara.repository.impl;

import edu.marmara.model.Advisor;
import edu.marmara.model.Instructor;
import edu.marmara.model.School;
import edu.marmara.repository.InstructorRepository;

import java.util.ArrayList;
import java.util.Objects;

public class InstructorRepositoryImpl implements InstructorRepository {
    // todo: Update the corresponding json file

    @Override
    public void save(Instructor instructor) {
        School school = School.getInstance();

        if (school.getInstructors() == null) {
            school.setInstructors(new ArrayList<>());
        }

        school.addInstructor(instructor);
    }

    @Override
    public Instructor findByEmail(String email) {
        School school = School.getInstance();

        if (school.getInstructors() == null) {
            return null;
        }
        for (Instructor instructor : school.getInstructors()) {
            if (Objects.equals(instructor.getEmail(), email)) {
                return instructor;
            }
        }
        return null;
    }

    @Override
    public Advisor findAdvisorByEmail(String advisor) {
        School school = School.getInstance();

        if (school.getInstructors() == null) {
            return null;
        }

        for (Instructor instructor : school.getInstructors()) {
            if (Objects.equals(instructor.getEmail(), advisor)) {
                if (instructor instanceof Advisor) {
                    return (Advisor) instructor;
                }
            }
        }

        return null;
    }
}

File: src/main/marmara/repository/impl/InstructorRepositoryImpl.py
from src.main.marmara.model.Advisor import Advisor
from src.main.marmara.model.School import School


class InstructorRepositoryImpl:
    def save(self, instructor):
        school = School.get_instance()
        if school.instructors is None:
            school.instructors = []
        school.add_instructor(instructor)

    def find_by_email(self, email):
        school = School.get_instance()
        if school.instructors is None:
            return None
        for instructor in school.instructors:
            if instructor.email == email:
                return instructor
        return None

    def find_advisor_by_email(self, advisor):
        school = School.get_instance()
        if school.instructors is None:
            return None
        for instructor in school.instructors:
            if instructor.email == advisor:
                if isinstance(instructor, Advisor):
                    return instructor
        return None
